packages/common/directives/src/filter-characters/filter-characters.directive.ts

```diff
@@ -0,0 +1,49 @@
+import { Directive, Input, OnDestroy, OnInit } from '@angular/core';
+import { AbstractControl } from '@angular/forms';
+import { filterCharacters } from '@angular-ru/common/utils';
+import { Subject } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+@Directive({ selector: '[filterCharacters]' })
+export class FilterCharactersDirective implements OnInit, OnDestroy {
+    @Input('filterCharacters')
+    public characters: string[] = [];
+
+    private control: AbstractControl | undefined;
+    private controlChanged$: Subject<boolean> = new Subject();
+
+    @Input()
+    public set formControl(control: AbstractControl | undefined) {
```

Comment:
why we need passed formControl?

instead 

```ts
constructor(private readonly ngControl: NgControl) {}
```

Reply:
because the controls may to change. we refused this realization and rewrote it as a showed in stackblitz.

Reply:
Schirbak suggested better solution by using ControlValueInterceptor